daemon/src/main/java/org/lsposed/lspd/service/LSPApplicationService.java

```diff
@@ -125,7 +126,20 @@ public ParcelFileDescriptor requestInjectedManagerBinder(List<IBinder> binder) t
             if (heartbeat != null) {
                 binder.add(ServiceManager.getManagerService().obtainManagerBinder(heartbeat, pid, uid));
             }
+        }
+        return ConfigManager.getInstance().getManagerApk();
+    }
 
+    @Override
+    public ParcelFileDescriptor requestCLIBinder(List<IBinder> binder) throws RemoteException {
+        ensureRegistered();
+        var pid = getCallingPid();
+        var uid = getCallingUid();
+        if (CLIValidator.managerStagePidValid(pid)) {
+            var heartbeat = handles.get(pid);
+            if (heartbeat != null) {
+                binder.add(ServiceManager.getManagerService().obtainManagerBinder(heartbeat, pid, uid));
+            }
         }
         return ConfigManager.getInstance().getManagerApk();
```

Comment:
why CLI needs the manager's apk fd

Reply:
In order to implement CLI with less time cost, a lot of code was copied from manager in my commit. Sorry for my carelessness.